src/services/ProblemService.ts

```diff
@@ -172,6 +174,44 @@ export class ProblemService {
     };
   }
 
+  public async solveProblem(userId: number, problemId: number) {
```

Comment:
nitpick: consider calling this as `markProblemAsSolved` since we may introduce an auto solving feature at some point and would like to reserve the name `solveProblem` at some point, helps in being explicit